modules/pool/lib/apr-data-sources/ib-tokens-apr.service.ts

```diff
@@ -0,0 +1,68 @@
+import { PoolAprService } from "../../pool-types";
+import { PrismaPoolWithExpandedNesting } from "../../../../prisma/prisma-types";
+import { prisma } from "../../../../prisma/prisma-client";
+import { networkContext } from "../../../network/network-context.service";
+import { prismaBulkExecuteOperations } from "../../../../prisma/prisma-util";
+import { PrismaPoolAprItemGroup } from "@prisma/client";
+import { TokenApr } from "./ib-yield-apr-handlers/types";
+import { IbYieldAprHandlers } from "./ib-yield-apr-handlers/ib-yield-apr-handlers";
+
+export class IbTokensAprService implements PoolAprService {
+  
+  private ibYieldAprHandlers: IbYieldAprHandlers;
+  
+  constructor(chainId: number) {
+    this.ibYieldAprHandlers = new IbYieldAprHandlers(chainId);
+  }
+
+  getAprServiceName(): string {
+    return "IbTokensAprService";
+  }
+
+  public async updateAprForPools(pools: PrismaPoolWithExpandedNesting[]): Promise<void> {
+    const operations: any[] = [];
+    const aprs = await this.fetchYieldTokensApr();
+    const tokenYieldPools = pools.filter((pool) => {
+        return pool.tokens.find((token) => {
+          return Array.from(aprs.keys()).map((key) => key.toLowerCase()).includes(token.address.toLowerCase());
+        })
+      }
+    );
+    for (const pool of tokenYieldPools) {
+      for (const token of pool.tokens) {
+        if ((aprs.get(token.address) !== undefined)) {
+          const tokenSymbol = token.token.symbol;
+          const itemId = `${ pool.id }-${ tokenSymbol }-yield-apr`
+
+          operations.push(prisma.prismaPoolAprItem.upsert({
+            where: { id_chain: { id: itemId, chain: networkContext.chain } },
+            create: {
+              id: itemId,
+              chain: networkContext.chain,
+              poolId: pool.id,
+              title: `${ tokenSymbol } APR`,
+              apr: aprs.get(token.address)?.val ?? 0,
+              group: (aprs.get(token.address)?.group as PrismaPoolAprItemGroup) ?? null,
+              type: pool.type === 'LINEAR' ? 'LINEAR_BOOSTED' : 'IB_YIELD',
```

Comment:
that is not enough, you'll need to check if it is the wrapped token when it is in a linear pool. only then it is linear_boosted, otherwise its IB_yield. E.g. for a wsteth/rfWSTETH linear pool